src/aeso_compiler.erl

```diff
@@ -211,13 +211,13 @@ check_call1(ContractString0, FunName, Args, Options) ->
                 %% First check the contract without the __call function
                 #{fcode := OrgFcode} = string_to_fcode(ContractString0, Options),
                 FateCode = aeso_fcode_to_fate:compile(OrgFcode, []),
-                _SymbolHashes = maps:keys(aeb_fate_code:symbols(FateCode)),
-                %% TODO collect all hashes and compute the first name without hash collision to
-                %% be used as __callX
-                %% case aeb_fate_code:symbol_identifier(<<"__call">>) of
-                ContractString = insert_call_function(ContractString0, FunName, Args, Options),
+                %% collect all hashes and compute the first name without hash collision to
+                SymbolHashes = maps:keys(aeb_fate_code:symbols(FateCode)),
+                CallName = first_none_match(?CALL_NAME, SymbolHashes,
+                                            lists:seq($1, $9) ++ lists:seq($A, $Z) ++ lists:seq($a, $z)),
+                ContractString = insert_call_function(ContractString0, CallName, FunName, Args, Options),
                 #{fcode := Fcode} = string_to_fcode(ContractString, Options),
-                #{args := CallArgs} = maps:get({entrypoint, <<"__call">>}, maps:get(functions, Fcode)),
+                #{args := CallArgs} = maps:get({entrypoint, list_to_binary(CallName)}, maps:get(functions, Fcode)),
```

Comment:
This will only work for nulllary functions (since the added function will always have 0 arguments...) This is a problem in the original PR not introduced here.